src/redux/books/books.js

```diff
@@ -1,26 +1,58 @@
+import * as api from '../../api/bookinfo';
+
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const GET_BOOKS = 'bookStore/books/GET_BOOKS';
+const CLEAR_BOOKS = 'bookStore/books/CLEAR_BOOKS';
 
 const initialState = [];
 
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+const handleData = (data) => {
+  const books = [];
+  const keys = Object.keys(data);
+  keys.forEach((key, index) => {
+    const book = data[key[index]];
+    book[0].item_id = key;
+
+    books.push(book[0]);
+  });
+};
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  id,
-});
+export const fetchBooks = () => async (dispatch) => api
+  .fetchBooks().then((data) => dispatch({ type: GET_BOOKS, payload: handleData(data) }));
+
```

Comment:
Your `removebook` method is not working as expected. If I have three(3) books, when I click on `remove` for one of the books, `the remove button will remove all the books available`. Kindly fix that.

Reply:
Alright, Thank you. Let me work on it